src/containers/InstructionsContainer.tsx

```diff
@@ -0,0 +1,158 @@
+        () => (
+          <HeadphonesDetectionScreen
+            headphoneType="ON_EAR"
+            onNext={onNextInstruction}
+          />
+        ),
+      ]
+    : []
+
+  // Append Volume Calibration if configured
+  function onFinishCalibration(volume) {
+    updateExperiment({ volume })
+    onNextInstruction()
+  }
+
+  if (mod.advancedVolumeCalibration) {
+    introScreens = introScreens.concat(() => (
+      <VolumeCalibrationScreen onFinishCalibration={onFinishCalibration} />
+    ))
+  }
+
+  // All slides for trual setup
+  const onNextTrialInstruction = () =>
+    setCurrentTrialInstruction(currentTrialInstruction + 1)
+  let trialScreens = mod.renderTrialTask
+    ? [
+        () => (
+          <TextInstructionScreen
+            heading="Task Instructions"
+            description="You will now see some images. You will be asked to rate your feelings and familiarity for these images."
+            actionLabel="Use the whole range of scales to help you make as detailed a rating of how you feel as possible."
+            color="teal"
+            onNext={onNextTrialInstruction}
+          />
+        ),
+        () => (
+          <RatingExplainationScreen
+            heading="A few seconds after each circle appears, this scale will disappear at the bottom of the screen."
+            description="Each time the scale appears, press the corresponding number on the screen to rate how much you expect to hear a scream."
+            color="teal"
+            onNext={onNextTrialInstruction}
+          />
+        ),
+        () => (
+          <RatingPracticeScreen
+            heading="Press any number to practice making a rating with the scaling below."
+            color="teal"
+            onNext={onNextTrialInstruction}
+          />
+        ),
+        () => (
+          <IntervalExplainationScreen
+            description="Before each circle is presented, you will see a white screen with a cross in the middle like the one shown above. It is important that you keep looking at the cross and wait for the next circle to appear."
+            color="teal"
+            onNext={onNextTrialInstruction}
+          />
+        ),
+        () => (
+          <TextInstructionScreen
+            heading="Instructions Complete"
+            description={`The experiment will now begin.\n\n You may occasionaly hear a scream.\n\n Remember to rate how much you expect to hear a scream by pressing a number every time the scale appears.`}
+            color="teal"
+            onNext={onNextTrialInstruction}
+          />
+        ),
+      ]
+    : []
+
+  // Get the correct screen
+  const IntroScreen = introScreens?.[currentAppInstruction]
+  const TrialScreen = trialScreens?.[currentTrialInstruction]
+  const CurrentScreen = IntroScreen ?? TrialScreen
+  const inRatingPhase = IntroScreen === undefined && TrialScreen !== undefined
+
+  // If no screens left then finish module
+  if (IntroScreen === undefined && TrialScreen === undefined) {
```

Comment:
Shouldn't this be in a `useEffect`? Not sure if we should be calling `onModuleComplete` in the render code path directly.

Reply:
Good Shout!